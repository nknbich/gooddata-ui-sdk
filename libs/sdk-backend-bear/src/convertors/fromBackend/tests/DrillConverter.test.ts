// (C) 2021 GoodData Corporation

import { convertDrillOrigin, convertVisualizationWidgetDrill } from "../DashboardConverter/index.js";
import {
    drillFromAttribute,
    drillFromMeasure,
    drillToDashboardWithDrillFromMeasure,
    drillToDashboardWithDrillFromAttribute,
} from "./DrillConverter.fixtures.js";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import * as uuid from "uuid";

describe("convert drill", () => {
    beforeEach(() => {
        vi.spyOn(uuid, "v4").mockReturnValue("mocked-uuid");
    });

    afterEach(() => {
        vi.clearAllMocks();
    });

    describe("origin", () => {
        it("should convert drill from measure", () => {
            const convertedDrill = convertDrillOrigin(drillFromMeasure);
            expect(convertedDrill).toMatchSnapshot();
        });

        it("should convert drill from attribute", () => {
            const convertedDrill = convertDrillOrigin(drillFromAttribute);
            expect(convertedDrill).toMatchSnapshot();
        });
    });

    describe("definition", () => {
        it("should convert drill to dashboard with drill from measure", () => {
            const convertedDrill = convertVisualizationWidgetDrill(drillToDashboardWithDrillFromMeasure);
            expect(convertedDrill).toMatchSnapshot();
        });

        it("should convert drill to dashboard with drill from attribute", () => {
            const convertedDrill = convertVisualizationWidgetDrill(drillToDashboardWithDrillFromAttribute);
            expect(convertedDrill).toMatchSnapshot();
        });
    });
});
