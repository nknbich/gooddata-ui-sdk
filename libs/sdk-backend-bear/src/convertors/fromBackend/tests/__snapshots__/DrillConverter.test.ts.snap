// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`convert drill > definition > should convert drill to dashboard with drill from attribute 1`] = `
{
  "localIdentifier": "mockeduuid",
  "origin": {
    "attribute": {
      "localIdentifier": "attributeLocalIdentifier",
    },
    "type": "drillFromAttribute",
  },
  "target": {
    "identifier": "dashboardId",
  },
  "transition": "in-place",
  "type": "drillToDashboard",
}
`;

exports[`convert drill > definition > should convert drill to dashboard with drill from measure 1`] = `
{
  "localIdentifier": "mockeduuid",
  "origin": {
    "measure": {
      "localIdentifier": "measureLocalIdentifier",
    },
    "type": "drillFromMeasure",
  },
  "target": {
    "identifier": "dashboardId",
  },
  "transition": "in-place",
  "type": "drillToDashboard",
}
`;

exports[`convert drill > origin > should convert drill from attribute 1`] = `
{
  "attribute": {
    "localIdentifier": "attributeLocalIdentifier",
  },
  "type": "drillFromAttribute",
}
`;

exports[`convert drill > origin > should convert drill from measure 1`] = `
{
  "measure": {
    "localIdentifier": "measureLocalIdentifier",
  },
  "type": "drillFromMeasure",
}
`;
